doc/Lifecycling.md

```diff
@@ -0,0 +1,223 @@
+# Lifecycling
```

Comment:
Please move this file to the root dir. `doc` is used for auto-generation of documentation and should be masked in `.gitignore`.

Reply:
Following @brianweir-da 's changes the automated doc has been moved to `.docs` and `docs` is used for non-automated documentation